lcls_tools/common/measurements/screen_profile.py

```diff
@@ -41,15 +40,18 @@ class ScreenBeamProfileMeasurementResult(BaseModel):
     rms_sizes: Optional[np.ndarray] = None
     centroids: Optional[np.ndarray] = None
     total_intensities: Optional[np.ndarray] = None
-    metadata: SkipValidation[SerializeAsAny[Any]]
+    metadata: SerializeAsAny[Any]
 
     model_config = ConfigDict(arbitrary_types_allowed=True, extra="forbid")
 
     @field_validator("*", mode="before")
     def validate_numpy_array(cls, v, field):
-        if v is not None:
-            if not isinstance(v, np.ndarray):
-                v = np.array(v)
+        """ convert all fields except metadata to numpy arrays """
+        if field.field_name != "metadata":
```

Comment:
This code is duplicated, same comments. More of an argument to encapsulate in the type.